R/image-interact-opts.R

```diff
@@ -0,0 +1,126 @@
+#'   then the hover coordinates will be available as \code{input$plot_hover}.
+#' @param delay How long to delay (in milliseconds) when debouncing or
+#'   throttling, before sending the mouse location to the server.
+#' @param delayType The type of algorithm for limiting the number of hover
+#'   events. Use \code{"throttle"} to limit the number of hover events to one
+#'   every \code{delay} milliseconds. Use \code{"debounce"} to suspend events
+#'   while the cursor is moving, and wait until the cursor has been at rest for
+#'   \code{delay} milliseconds before sending an event.
+#' @param clip Should the hover area be clipped to the plotting area? If FALSE,
+#'   then the server will receive hover events even when the mouse is outside
+#'   the plotting area, as long as it is still inside the image.
+#' @export
+hoverOpts <- function(id = NULL, delay = 300,
+                         delayType = c("debounce", "throttle"), clip = TRUE) {
+  if (is.null(id))
+    stop("id must not be NULL")
+
+  list(
+    id = id,
+    delay = delay,
+    delayType = match.arg(delayType),
+    clip = clip
+  )
+}
+
+#' Create an object representing brushing options
+#'
+#' This generates an object representing brushing options, to be passed as the
+#' \code{brush} argument of \code{\link{imageOutput}} or
+#' \code{\link{plotOutput}}.
+#'
+#' @param id Input value name. For example, if the value is \code{"plot_brush"},
+#'   then the coordinates will be available as \code{input$plot_brush}.
+#' @param color Fill color of the brush.
+#' @param outline Outline color of the brush.
+#' @param opacity Opacity of the brush
+#' @param delay How long to delay (in milliseconds) when debouncing or
+#'   throttling, before sending the brush data to the server.
+#' @param delayType The type of algorithm for limiting the number of brush
+#'   events. Use \code{"throttle"} to limit the number of brush events to one
+#'   every \code{delay} milliseconds. Use \code{"debounce"} to suspend events
+#'   while the cursor is moving, and wait until the cursor has been at rest for
+#'   \code{delay} milliseconds before sending an event.
+#' @param clip Should the brush area be clipped to the plotting area? If FALSE,
+#'   then the user will be able to brush outside the plotting area, as long as
+#'   it is still inside the image.
+#' @param direction The direction for brushing. If \code{"xy"}, the brush can be
+#'   drawn and moved in both x and y directions. If \code{"x"}, or \code{"y"},
+#'   the brush wil work horizontally or vertically.
+#' @param resetOnNew When a new image is sent to the browser (via
+#'   \code{\link{renderImage}}), should the brush be reset? The default,
+#'   \code{FALSE}, is useful if you want to update the plot while keeping the
+#'   brush. Using \code{TRUE} is useful if you want to clear the brush whenever
+#'   the plot is updated.
+#' @export
+brushOpts <- function(id = NULL, color = "#666", outline = "#000",
```

Comment:
Sure, stroke and fill sound good - it's also consistent with ggvis terminology.

As for independent opacity of stroke, it's a little tricky since it requires using `rgba()` syntax, and this will require translating from color names to RGB values. Also, IE8 apparently doesn't support `rgba()`, so we'll have to add a workaround for that.